checker/src/abstract_value.rs

```diff
@@ -2884,6 +2890,35 @@ impl AbstractValueTrait for Rc<AbstractValue> {
                     default.clone()
                 }
             }
+            Expression::UnknownTagCheck {
+                path,
+                tag,
+                checking_presence,
+            } => {
+                if let Some(val) = environment.value_at(&path) {
+                    let tag_presence = val.has_tag(tag);
+                    if tag_presence.is_top() {
+                        // Still unknown, fall through.
```

Comment:
Instead of falling through you could just put the UnknownTagCheck expression here.

Reply:
In this block there are two cases where we return UnknownTagCheck: either path is not in the environment, or tag_presence is top. Should we duplicate the creation of the UnknownTagCheck expression?

Reply:
Missed that. Leave it as is then.